src/Components/Sidebar/Sidebar.js

```diff
@@ -0,0 +1,321 @@
+import React, { useState, useEffect, useRef } from "react";
+import './Sidebar.css'
+import { useProductAvailable } from "../../Context/product-context"
+import { useGenre } from "../../Context/genre-context";
+
+function Sidebar() {
+  const {
+  productsAvailableList,
+  dispatchSortedProductsList,
+  productFilterOptions,
+  dispatchProductFilterOptions
+  } = useProductAvailable()
+
+  const {
+    fictionCategoryCheckbox,
+    setFictionCategoryCheckbox,
+    thrillerCategoryCheckbox, 
+    setThrillerCategoryCheckbox,
+    techCategoryCheckbox, 
+    setTechCategoryCheckbox,
+    philosophyCategoryCheckbox, 
+    setPhilosophyCategoryCheckbox,
+    romanceCategoryCheckbox, 
+    setRomanceCategoryCheckbox,
+    mangaCategoryCheckbox, 
+    setMangaCategoryCheckbox, 
+  } = useGenre()
+
+  const ratingRadioBtnRef = useRef(null)
+  
```

Comment:
Confusion: Why ref is needed here??

Reply:
Oh, that is because, when user click Clear Filters, I use ref to update radio button to default.